From e46b9858a3c5138998d4cf118797f9ca2ef64ba9 Mon Sep 17 00:00:00 2001
From: "Subhransu S. Prusty" <subhransu.s.prusty@intel.com>
Date: Wed, 20 Aug 2014 17:39:28 +0530
Subject: Audio: mofd: platform: update the mask bit in sst_mix_put

Issue : when same register is updated, changing one alsa control
was resulting in wrong update for other alsa control bit resiging
in the same register.
This is because in sst_mix_put soc_dapm_mixer_update_power is called which
inturn calls  soc_widget_update_bits_locked but the mask expected there is
the bit to be updated but the mix_put was always sending 1 as mask
hence it was causing issues, so update the mask accordingly based on the shift.

Change-Id: Ic03caf0cd425c49a2b5d362fe8908f6b63736c54
Signed-off-by: Subhransu S. Prusty <subhransu.s.prusty@intel.com>
---
 sound/soc/intel/platform-libs/controls_v2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/intel/platform-libs/controls_v2.c b/sound/soc/intel/platform-libs/controls_v2.c
index 0528384..b1d9702 100644
--- a/sound/soc/intel/platform-libs/controls_v2.c
+++ b/sound/soc/intel/platform-libs/controls_v2.c
@@ -128,7 +128,7 @@ int sst_mix_put(struct snd_kcontrol *kcontrol,
 	dapm_kcontrol_set_value(kcontrol, val);
 	update.kcontrol = kcontrol;
 	update.reg = mc->reg;
-	update.mask = mask;
+	update.mask = mask << mc->shift;
 	update.val = val;
 
 	snd_soc_dapm_mixer_update_power(widget->dapm, kcontrol, connect, &update);
-- 
2.1.0

